tests/dataset/test_aoi.py

```diff
@@ -81,6 +81,65 @@ def test_bounds_iou(self) -> None:
         expected_iou = 0.013904645827033404
         assert iou == expected_iou
 
+    def test_corrupt_raster(self) -> None:
+        raster_file = "tests/data/massachusetts_buildings_kaggle/22978945_15_uint8_clipped_clipped.tif"
+        with pytest.raises(BaseException):
+            raster = rasterio.open(raster_file)
+
+    def test_image_only(self) -> None:
+        extract_archive(src="tests/data/spacenet.zip")
+        data = read_csv("tests/sampling/sampling_segmentation_binary-multiband_ci.csv")
+        for row in data:
+            aoi = AOI(raster=row['tif'], label=None)
+
+    def test_missing_label(self) -> None:
+        extract_archive(src="tests/data/spacenet.zip")
+        data = read_csv("tests/sampling/missing_label.csv")
+        for row in data:
+            with pytest.raises(rasterio.errors.RasterioIOError):
+                aoi = AOI(raster=row['tif'], label=row['gpkg'], split=row['split'])
+
+    def test_missing_raster(self) -> None:
+        extract_archive(src="tests/data/spacenet.zip")
+        data = read_csv("tests/sampling/missing_raster.csv")
+        for row in data:
+            with pytest.raises(rasterio.errors.RasterioIOError):
+                aoi = AOI(raster=row['tif'], label=row['gpkg'], split=row['split'])
+
+    def test_wrong_split(self) -> None:
+        extract_archive(src="tests/data/spacenet.zip")
+        data = read_csv("tests/sampling/wrong_split.csv")
+        for row in data:
+            with pytest.raises(ValueError):
+                aoi = AOI(raster=row['tif'], label=row['gpkg'], split=row['split'])
+
+    def test_wrong_seperation(self) -> None:
```

Comment:
The two following tests seem ok, but should be in a separate script since they are testing, in reality the read_csv() function, which is in utils/utils.py. These tests should therefore be in tests/utils/test_utils.py